dallinger/config.py

```diff
@@ -86,7 +86,6 @@ def is_valid_json(value):
     ("prolific_estimated_completion_minutes", int, []),
     ("prolific_maximum_allowed_minutes", int, []),
     ("prolific_recruitment_config", six.text_type, [], False, [is_valid_json]),
-    ("prolific_reward_cents", int, []),
```

Comment:
I know it's annoying but I think we should throw a useful error message if someone accidentally uses `prolific_reward_cents` in their config. Otherwise people could have some serious unexpected bugs.

Reply:
The removal of `prolific_reward_cents` will be a breaking change and be announced in the CHANGELOG, while also resulting in a new major version. I don't feel comfortable with adding code for things that don't exist anymore. Normally, instead of suddenly removing  `prolific_reward_cents`  we should rather deprecate it first, warning strongly against its use. Your thoughts?

Reply:
There is a spectrum of how nice we can be as programmers:

- **Nasty**
- Change it without telling anyone
- Change it and say so in the CHANGELOG
- Change it and give people a helpful error message if they don't realise it's been changed
- Deprecate it and have a period where you support both old and new functionality
- **Nice**

The 'deprecation' option is nicest but takes the most work because you have to implement dual functionality. This is harder and more error-prone than just renaming the variables.

Change it and give a helpful error message is a good compromise in my opinion. It is very easy to give a helpful error message here. Simply add a validation method when the config is loaded that checks whether `prolific_reward_cents` is not `None`, and if so, throw a nice error message. I don't see that this is a problem with 'adding code for things that don't exist anymore', this is simply us adding a straightforward validation function.

Reply:
OK, convinced. I added a custom `KeyError` message now. Although, without checking for the value of `None`. You don't want to have it set at all, do you?

Reply:
Yep, sounds good